src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -374,1042 +376,665 @@
 
 // Mixins
-import { DateMixin, EnumMixin, FilingMixin, LegalApiMixin } from '@/mixins'
-
-export default {
-  name: 'FilingHistoryList',
-
-  mixins: [DateMixin, EnumMixin, FilingMixin, LegalApiMixin],
+import { DateMixin, EnumMixin, FilingMixin, LegalApiMixin, PayApiMixin } from '@/mixins'
 
+@Component({
   components: {
     // sub-components
-    ColinFiling,
     CompletedAlteration,
     CompletedIa,
+    FiledLabel,
     FutureEffective,
     FutureEffectivePending,
     PaperFiling,
     PendingFiling,
+    StaffFiling,
+
     DetailsList,
+
     // dialogs
     AddCommentDialog,
     DownloadErrorDialog,
     LoadCorrectionDialog
   },
-
-  data () {
-    return {
-      addCommentDialog: false,
-      downloadErrorDialog: false,
-      loadCorrectionDialog: false,
-      panel: null as number, // currently expanded panel
-      historyItems: [] as Array<HistoryItemIF>,
-      loadingDocument: false,
-      loadingReceipt: false,
-      loadingAll: false,
-      currentFilingId: null as number,
-      downloadingDocIndex: -1,
-      FilingTypes // enum for template
-    }
-  },
-
-  props: {
-    disableChanges: null
-  },
-
   computed: {
+    // Property definitions for runtime environment.
     ...mapGetters(['isBComp', 'isRoleStaff', 'nrNumber']),
+    ...mapState(['filings', 'entityName'])
+  }
+})
+export default class FilingHistoryList extends Mixins(
+  DateMixin, EnumMixin, FilingMixin, LegalApiMixin, PayApiMixin
+) {
+  @Prop({ default: false })
+  readonly disableChanges: boolean
+
+  // enums for template
+  readonly DocumentTypes = DocumentTypes
+  readonly FilingTypes = FilingTypes
+
+  readonly isBComp!: boolean
+  readonly isRoleStaff!: boolean
+  readonly nrNumber!: string
+  readonly filings!: Array<LedgerIF>
+  readonly entityName!: string
+
+  // local properties
+  private addCommentDialog = false
+  private downloadErrorDialog = false
+  private loadCorrectionDialog = false
+  private panel: number = null // currently expanded panel
+  private historyItems: Array<HistoryItemIF> = []
+  private loadingDocument = false
+  private loadingReceipt = false
+  private loadingAll = false
+  private currentFilingId: number = null
+  private downloadingDocIndex = -1
+  private isBusy = false
+
+  /** The Edit URL string. */
+  private get editUrl (): string {
+    return sessionStorage.getItem('EDIT_URL')
+  }
+
+  /** The IA's Temporary Registration Number string. */
+  private get tempRegNumber (): string {
+    return sessionStorage.getItem('TEMP_REG_NUMBER')
+  }
+
+  private created (): void {
+    // load data into this page
+    this.loadData()
+  }
 
-    ...mapState(['filings', 'entityIncNo', 'entityName', 'entityType']),
+  private loadData (): void {
+    this.historyItems = []
 
-    /** The Pay API URL string. */
-    payApiUrl (): string {
-      return sessionStorage.getItem('PAY_API_URL')
-    },
+    // create history items from 'filings' array from API
```

Comment:
I considered using just one list (ie, the original ledgers/filings array from the API) instead of creating a second array -- and I synced the property names between them to be closer to that -- but it would be overly complicated... so there are still 2 arrays.